service/src/main/java/tds/config/repositories/impl/ConfigRepositoryImpl.java

```diff
@@ -58,4 +65,128 @@ public ConfigRepositoryImpl(NamedParameterJdbcTemplate jdbcTemplate) {
 
         return clientSystemFlags;
     }
+
+    @Override
+    public Optional<ClientSystemMessage> findClientSystemMessage(String clientName, String messageKey, String language, String clientDefaultLanguage, String context, String subject, String grade) {
+        List<String> langauges = Arrays.asList(language, clientDefaultLanguage);
```

Comment:
If you think that check should be done I actually think I would do it in the service not the repo.